### DISCUSSION TITLE
Can't have VsCode breakpoints to work in SSR endpoints

### DISCUSSION THREAD

**Discussion by IvankoB** [sveltejs/kit#6374] [Q&A]
### Describe the bug

Me failed to get stepped debugging SSR code of the default (vite-based, created in the way recommended on the SvelteKit site) project in VsCode. Many settings in 'vite.config.js' & 'launch.js' were tried.

### Reproduction

Any project with SSR endpoints

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (4) x64 Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz
    Memory: 1.27 GB / 11.90 GB
  Binaries:
    Node: 16.16.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.17.0 - D:\MyProjects\JS\SVELTEKIT\pokemon-tutorial\node_modules\.bin\npm.CMD
  Browsers:
    Chrome: 104.0.5112.102
    Edge: Spartan (44.22000.120.0), Chromium (104.0.1293.70)
    Internet Explorer: 11.0.22000.120
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.64
    @sveltejs/kit: next => 1.0.0-next.445
    svelte: ^3.44.0 => 3.49.0
    vite: ^3.0.0 => 3.0.9
```

### Severity

serious, but I can work around it

**Comment by dummdidumm** [comment]
I don't think this is something we can "fix", this sounds more like a usage question, therefore turning this into a discussion.

**Comment by thenbe** [comment]
In dev, you can get vscode step debugging on the server like so:
1. Use a [`debugger`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/debugger) statement.
2. Vscode command palette > attach to running node process
3. The program will pause when you hit the debugger statement.

But if you use `adapter-node` do `npm run build && node --inspect -r source-map-support/register build/index.js`, you can get step debugging without using a manual `debugger` statement, but you'll have to set up your sourcemaps.

Here is a vscode launch config, but remember to run `npm run build` first. You can also add it as a [preLaunch](https://code.visualstudio.com/Docs/editor/debugging#_launchjson-attributes) task.
```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "debug sveltekit server",
      "type": "node",
      "request": "launch",
      "cwd": "${workspaceFolder}",
      "skipFiles": [
        "<node_internals>/**"
      ],
      "args": [
        "--inspect",
        "-r",
        "source-map-support/register",
        "build/index.js"
      ],
      "console": "internalConsole"
    }
  ]
}
```

More info here: https://www.youtube.com/watch?v=ubeOpB9FlZE

**Comment by IvankoB** [comment]
Thanx !
